ecs_logging/_utils.py

```diff
@@ -111,8 +111,16 @@ def merge_dicts(from_, into):
     When called has side-effects within 'destination'.
     """
     for key, value in from_.items():
-        if isinstance(value, dict):
-            merge_dicts(value, into.setdefault(key, {}))
+        into.setdefault(key, {})
+        if isinstance(value, dict) and isinstance(into[key], dict):
+            merge_dicts(value, into[key])
+        elif into[key]:
```

Comment:
This boolean check here worries me, for example what if `into[key]` is something falsy like `0` or the empty string?

Reply:
I think in that case

```
elif into[key] != {}:
```

Because that's effectively checking `if key in into:` after the `setdefault` above.